lib/actions/config.js

```diff
@@ -0,0 +1,417 @@
+
+		return Promise.try(
+			() => options.drive || getVisuals().drive('Select the device drive'),
+		)
+			.tap(umountAsync)
+			.then(drive => config.read(drive, options.type))
+			.tap(configJSON => {
+				console.info(prettyjson.render(configJSON));
+			});
+	},
+};
+
+export const write = {
+	signature: 'config write <key> <value>',
+	description: 'write a device configuration',
+	help: `\
+Use this command to write the config.json file to the mounted filesystem (e.g. SD card) of a provisioned device
+
+Examples:
+
+	$ balena config write --type raspberry-pi username johndoe
+	$ balena config write --type raspberry-pi --drive /dev/disk2 username johndoe
+	$ balena config write --type raspberry-pi files.network/settings "..."\
+`,
+	options: [
+		{
+			signature: 'type',
+			description:
+				'device type (Check available types with `balena devices supported`)',
+			parameter: 'type',
+			alias: 't',
+			required: 'You have to specify a device type',
+		},
+		{
+			signature: 'drive',
+			description: 'drive',
+			parameter: 'drive',
+			alias: 'd',
+		},
+	],
+	permission: 'user',
+	root: true,
+	action(params, options) {
+		const Promise = require('bluebird');
+		const _ = require('lodash');
+		const config = require('balena-config-json');
+		const umountAsync = Promise.promisify(require('umount').umount);
+
+		return Promise.try(
+			() => options.drive || getVisuals().drive('Select the device drive'),
+		)
+			.tap(umountAsync)
+			.then(drive =>
+				config
+					.read(drive, options.type)
+					.then(function(configJSON) {
+						console.info(`Setting ${params.key} to ${params.value}`);
+						_.set(configJSON, params.key, params.value);
+						return configJSON;
+					})
+					.tap(() => umountAsync(drive))
+					.then(configJSON => config.write(drive, options.type, configJSON)),
+			)
+			.tap(() => {
+				console.info('Done');
+			});
+	},
+};
+
+export const inject = {
+	signature: 'config inject <file>',
+	description: 'inject a device configuration file',
+	help: `\
+Use this command to inject a config.json file to the mounted filesystem
+(e.g. SD card or mounted balenaOS image) of a provisioned device"
+
+Examples:
+
+	$ balena config inject my/config.json --type raspberry-pi
+	$ balena config inject my/config.json --type raspberry-pi --drive /dev/disk2\
+`,
+	options: [
+		{
+			signature: 'type',
+			description:
+				'device type (Check available types with `balena devices supported`)',
+			parameter: 'type',
+			alias: 't',
+			required: 'You have to specify a device type',
+		},
+		{
+			signature: 'drive',
+			description: 'drive',
+			parameter: 'drive',
+			alias: 'd',
+		},
+	],
+	permission: 'user',
+	root: true,
+	action(params, options) {
+		const Promise = require('bluebird');
+		const util = require('util');
+		const config = require('balena-config-json');
+		const umountAsync = Promise.promisify(require('umount').umount);
+		const readFileAsync = util.promisify(require('fs').readFile);
```

Comment:
Switched this to `util.promisify` as it gives much better typing - available from 8.0.0 onwards so matches our supported versions